src/WidgetIncluderModule.php

```diff
@@ -21,53 +21,64 @@ public function main(App $app)
 
     public function renderWidget(App $app, $widget_id, $options = [], $view = null)
     {
+
+        // get user context
         $user = $app->user();
-        
+
         /** @var Widget $widget */
         if (!$widget = Widget::where([
-            'id' => $widget_id
+            'id' => $widget_id,
         ])->first()) {
             throw new App\Exception('Widget not found', 404);
         }
-        
-        if (!$widget->hasAccess(App::user())
-                or !$type = App::widget($widget->type)
-                or $widget->status !== 1) {
+
+        // widget type
+        $type = App::widget($widget->type);
+
+        // check permissions
+        if ($widget->hasAccess($user) !== true
+            || $type === false
```

Comment:
Just found out that this will not work. The value of `$type` is `null` if pagekit can not find a matching value for the type stored for the widget. So this comparison leads to a dump if there is a unknown type. Replacing `false` with `null` solves this.